controller/contour/controller.go

```diff
@@ -103,26 +103,47 @@ func (r *Reconciler) SetupWithManager(mgr ctrl.Manager) error {
 
 // ensureContour ensures all necessary resources exist for the given contour
 // in namespace ns.
-func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour, ns string) error {
+func (r *Reconciler) ensureContour(ctx context.Context, contour *operatorv1alpha1.Contour) error {
 	// Before doing anything with the contour, ensure it has a finalizer
 	// so it can cleaned-up later.
 	if err := r.ensureFinalizer(ctx, contour); err != nil {
 		return fmt.Errorf("failed to finalize contour %s/%s: %w", contour.Namespace, contour.Name, err)
 	}
-	if err := r.ensureNamespace(ctx, ns); err != nil {
-		return fmt.Errorf("failed to ensure namespace %s: %w", ns, err)
+	if err := r.ensureNamespace(ctx, contour); err != nil {
+		return fmt.Errorf("failed to ensure namespace %s for contour %s/%s: %w",
+			contour.Spec.Namespace.Name, contour.Namespace, contour.Name, err)
 	}
 	return nil
 }
 
 // ensureContourRemoved ensures all resources for the given contour do not exist
 // in namespace ns.
```

Comment:
nit: the godoc comment still refers to `ns`.

Reply:
ack